frontend/src/ui/Floating.tsx

```diff
@@ -29,6 +29,7 @@ import { bug, unreachable } from "../util/err";
 
 type Context = {
     open: boolean;
+    setOpen: React.Dispatch<React.SetStateAction<boolean>> | null;
```

Comment:
This is fine but you could also just use `setOpen: null | (v: boolean) => void);`